otwcode/otwarchive
# AO3-4008 Point comments links directly to comments section

# Pull Request Checklist

Partly. Added some supportive tests, but couldn't add tests to directly check scrolling. Attempting to bypass Capybara and test "what is visible on screen at the current moment" doesn't seem like a very stable automated test.

## Issue

https://otwarchive.atlassian.net/browse/AO3-4008

## Purpose

Attempts to make comment links point to the comments section (not feedback one, not kudos one).
- Note that if there are no comments (or comments are not yet loaded) on big screens, scroll will happen to the bottom of the page because comments section is too low down. So it might seem as if nothing changed. On small window sizes difference should however be visible. 
- When loading with Top comments button the same behavior would be observed, but then the comments will get added shortly after, thus expanding the space below.

I am still not very certain about what I am doing though :smile:  

## Testing Instructions

Comment links are now supposed to link user to "Comments" section, where the top element is Add Comment form (or comments notice if form is not available).



|                                                | Before: work with no comments | After: work with no comments | Before: work with comments                   | After: work with comments                       |
|------------------------------------------------|-------------------------------|------------------------------|----------------------------------------------|-------------------------------------------------|
| Top Comments Button                            | Kudos section                 | Comments section             | Feedback section                             | Comments section                                |
| Down Comments Button                           | -                             | -                            | JS enabled: NONE<br>JS disabled: Kudos section | JS enabled: NONE<br>JS disabled: Comments section |
| [Direct link with show_comments](http://localhost:3000/works/1?show_comments=true#comments) | Kudos section                 | Comments section             | Kudos section                                | Comments section                                |
| [Direct link without show_comments](http://localhost:3000/works/1#comments)              | Kudos section                 | Comments section             | Kudos section                                | Comments section                                |
| From works blurb: Comments link                | -                             | -                            | Kudos section                                | Comments section                                |
| From works blurb: Kudos link                   | Kudos section                 | Kudos section                | Kudos section                                | Kudos section                                   |

Please note that *screen size matters*
Effects should be visible on small-sized windows, possibly best on phones
For big screens users will still be scrolled somewhere near "kudos" section because it's impossible to scroll further down.

No change in behavior regarding expanded/collapsed comments.

**Note**: with disabled javascript "Hide Comments" buttons redirect to comments section. That doesn't happen when javascript is enabled. No changes made about that. 

## Credit

korrien, she/her
